src/downloadHns.js

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+const fs = require("fs");
+
+const { DEFAULT_DOWNLOAD_HNS_OPTIONS } = require("./defaults");
+
+const downloadFileHns = async function (path, domain, customOptions = {}) {
```

Comment:
I would keep this in `download.js`. Also, this function should correspond to the implementation of `downloadFile`, i.e. the signature should start with

```js
SkynetClient.prototype.downloadFileHns = ...
```

and it should contain

```js
const writer = fs.createWriteStream(path);
```